Dockerfile

```diff
@@ -1,17 +1,20 @@
 ##########################################################
 # Core Mava image
-FROM registry.kao.instadeep.io/library/nvidia/cuda:11.4.2-cudnn8-runtime-ubuntu20.04 as mava-core
+FROM nvidia/cuda:11.4.2-cudnn8-runtime-ubuntu20.04 as mava-core
 # Flag to record agents
 ARG record
 # Ensure no installs try launch interactive screen
 ARG DEBIAN_FRONTEND=noninteractive
 # Update packages
-RUN apt-get update -y && apt-get install -y python3-pip
+RUN apt-get update -y && apt-get install -y python3-pip && apt-get install -y python3-venv
 # Update python path
 RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 10 &&\
     rm -rf /root/.cache && apt-get clean
-# Upgrade pip
-RUN python -m pip install --upgrade pip
```

Comment:
I like that we're moving away from having a global env in the container to a virtual env.

Just double checking that this won't cause problems with how we run anything? Off the top of my head, it should all be fine because the `PATH` includes the venv, but I'm concerned that this will break some fringe way we have of running things.

Reply:
I have tested the mava tf core, pz and flatland images (the images impacted by this change). They all seem to run and work with the new changes :+1: